Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/StateSynchronization/SpatialLocalization/HostCoordinateLocalizationMechanism.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Experimental.SpatialAlignment.Common;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.Experimental.SpectatorView
+{
+    /// <summary>
+    /// Helper localization method here the host finds a QR code and gives it to the observer
+    /// </summary>
+    internal abstract class HostCoordinateLocalizationMechanism : LocalizationMechanismBase
```

Comment:
>HostCoordinateLocalizationMechanism [](start = 28, length = 35)

I think that we should stick to broadcaster/observer nomenclature. Host is a new term that just increases the cognitive load/may lead to confusion.

Reply:
Granted, it seems this is mainly a helper class for both types. Would it better if this was split into two classes?

---
In reply to: [285283658](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/4460#discussion_r285283658) [](ancestors = 285283658)

Reply:
I will experiment with this more in the future.

---
In reply to: [285284098](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/4460#discussion_r285284098) [](ancestors = 285284098,285283658)